integration-libs/epd-visualization/components/visual-picking/visual-picking-tab/product-list/visual-picking-product-list.service.ts

```diff
@@ -0,0 +1,171 @@
+import { EventEmitter, Injectable } from '@angular/core';
+import {
+  Product,
+  ProductReference,
+  ProductReferenceService,
+} from '@spartacus/core';
+import { CurrentProductService } from '@spartacus/storefront';
+import { combineLatest, Observable } from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  shareReplay,
+  switchMap,
+  tap,
+} from 'rxjs/operators';
+import { VisualPickingProductListItem } from './model/visual-picking-product-list-item.model';
+import { VisualPickingProductFilterService } from '../product-filter/visual-picking-product-filter.service';
+import {
+  EpdVisualizationConfig,
+  EpdVisualizationInnerConfig,
+  VisualPickingConfig,
+} from '@spartacus/epd-visualization/root';
+
+@Injectable({
+  providedIn: 'any',
+})
+export class VisualPickingProductListService {
+  constructor(
+    protected currentProductService: CurrentProductService,
+    protected productReferenceService: ProductReferenceService,
+    protected visualPickingProductFilterService: VisualPickingProductFilterService,
+    protected epdVisualizationConfig: EpdVisualizationConfig
+  ) {}
+
+  protected readonly DEFAULT_ITEMS_PER_SLIDE = 7;
+
+  /**
+   * Initializes the service.
+   */
+  public initialize(): void {
+    this.getFilteredProductReferences$().subscribe(() => {
+      this.activeSlideStartIndex = 0;
+    });
+
+    this.filteredItems$.subscribe((items) => {
```

Comment:
To avoid memory leaks, every subscription needs to be unsubscibed, at least in [`ngOnDestroy()` lifecycle hook](https://angular.io/api/core/OnDestroy). Please double-check also other files in your PR if any subscriptions need unsubscription.

Reply:
Yes. This is something I have been working on. I will implement unsubscription.